src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAddressService.java

```diff
@@ -55,7 +65,57 @@ public AdminCourtAddressService(final CourtRepository courtRepository, final Cou
             .collect(toList());
     }
 
-    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    public List<String> validateCourtAddressPostcodes(final String slug, final List<CourtAddress> courtAddresses) {
```

Comment:
the name for the method `validateCourtAddressPostcodes` does not seem to indicate that it is also updating values via lines 82 `updateCourtLatLonUsingPrimaryPostcode`, so may need a slight tweak